fastai/learner.py

```diff
@@ -167,9 +187,17 @@ def fit_gen(self, model, data, layer_opt, n_cycle, cycle_len=None, cycle_mult=1,
 
         if best_save_name is not None:
             callbacks+=[SaveBestModel(self, layer_opt, metrics, best_save_name)]
+            
+        if use_swa:
+            # make a copy of the model to track average weights
+            self.swa_model = copy.deepcopy(model)
```

Comment:
I'm not sure about using `deepcopy`, in terms of either correctness or efficiency.  Would `state_dict()` be better? Or grab the params from layer_opt?